src/OptionsStruct.jl

```diff
@@ -1,9 +1,75 @@
 module OptionsStructModule
 
 using Optim: Optim
+using StatsBase: StatsBase
+import Random: AbstractRNG
 import DynamicExpressions: AbstractOperatorEnum
 import LossFunctions: SupervisedLoss
 
+mutable struct MutationWeights
+    mutate_constant::Float64
+    mutate_operator::Float64
+    add_node::Float64
+    insert_node::Float64
+    delete_node::Float64
+    simplify::Float64
+    randomize::Float64
+    do_nothing::Float64
+end
+
+const mutations = fieldnames(MutationWeights)
+
+"""
+    MutationWeights(;kws...)
+
+This defines how often different mutations occur. These weightings
+will be normalized to sum to 1.0 after initialization.
+# Arguments
+- `mutate_constant::Float64`: How often to mutate a constant.
+- `mutate_operator::Float64`: How often to mutate an operator.
+- `add_node::Float64`: How often to append a node to the tree.
+- `insert_node::Float64`: How often to insert a node into the tree.
+- `delete_node::Float64`: How often to delete a node from the tree.
+- `simplify::Float64`: How often to simplify the tree.
+- `randomize::Float64`: How often to create a random tree.
+- `do_nothing::Float64`: How often to do nothing.
+"""
+function MutationWeights(;
+    mutate_constant=0.048,
+    mutate_operator=0.47,
+    add_node=0.79,
+    insert_node=5.1,
+    delete_node=1.7,
+    simplify=0.0020,
+    randomize=0.00023,
+    do_nothing=0.21,
+)
+    return MutationWeights(
+        mutate_constant,
+        mutate_operator,
+        add_node,
+        insert_node,
+        delete_node,
+        simplify,
+        randomize,
+        do_nothing,
+    )
+end
+
+"""Convert MutationWeights to a vector."""
+@generated function Base.convert(
+    ::Type{V}, weightings::MutationWeights
+) where {V<:AbstractVector}
+    fields = [:(weightings.$(mut)) for mut in mutations]
+    return :(V([$(fields...)]))
+end
+
+"""Sample a mutation, given the weightings."""
+function Base.rand(rng::AbstractRNG, weightings::MutationWeights)::Symbol
```

Comment:
Instead of `rand`, this should be `sample_mutation`. Should have `mutations` (i.e., the constant) and `weightings` as arguments. While this isn't necessary I think it would be nicer.